tests/integration_tests_rust.rs

```diff
@@ -271,26 +271,45 @@ fn start_stop_reinit() {
 #[test]
 fn onchain_spend_receive() {
 	let (bitcoind, electrsd) = setup_bitcoind_and_electrsd();
-	let chain_source = TestChainSource::Esplora(&electrsd);
+	let chain_source = TestChainSource::BitcoindRpc(&bitcoind);
 	let (node_a, node_b) = setup_two_nodes(&chain_source, false, true, false);
 
 	let addr_a = node_a.onchain_payment().new_address().unwrap();
 	let addr_b = node_b.onchain_payment().new_address().unwrap();
 
+	let premine_amount_sat = 1_100_000;
 	premine_and_distribute_funds(
 		&bitcoind.client,
 		&electrsd.client,
-		vec![addr_b.clone()],
-		Amount::from_sat(100000),
+		vec![addr_a.clone(), addr_b.clone()],
+		Amount::from_sat(premine_amount_sat),
 	);
 
 	node_a.sync_wallets().unwrap();
 	node_b.sync_wallets().unwrap();
-	assert_eq!(node_b.list_balances().spendable_onchain_balance_sats, 100000);
+	assert_eq!(node_a.list_balances().spendable_onchain_balance_sats, premine_amount_sat);
+	assert_eq!(node_b.list_balances().spendable_onchain_balance_sats, premine_amount_sat);
+
+	let channel_amount_sat = 1_000_000;
+	let reserve_amount_sat = 25_000;
+	open_channel(&node_b, &node_a, channel_amount_sat, true, &electrsd);
+	generate_blocks_and_wait(&bitcoind.client, &electrsd.client, 6);
+
+	node_a.sync_wallets().unwrap();
+	node_b.sync_wallets().unwrap();
+	expect_channel_ready_event!(node_a, node_b.node_id());
+	expect_channel_ready_event!(node_b, node_a.node_id());
+
+	let node_a_balance = premine_amount_sat - reserve_amount_sat;
+	let node_b_balance_lower = premine_amount_sat - channel_amount_sat - reserve_amount_sat - 1000;
```

Comment:
Is this because `1000` sats is later sent? Might be worth making a constant for `1000` or adding a comment.